src/components/TodoModal/Dropdown.tsx

```diff
@@ -1,70 +1,139 @@
-import { Goal, Todo } from '@/types/interface';
+import { Goal } from '@/types/interface';
 import { ArrowDropdownIcon } from '@assets';
+import useGetGoals from '@hooks/api/goalsAPI/useGetGoals';
+import { useEffect, useRef, useState } from 'react';
+import { useInView } from 'react-intersection-observer';
 
 interface DropdownProps {
-  options: Goal[];
-  selectedOption: Todo['goal'] | null;
-  onSelect: (goal: Todo['goal'] | null) => void;
+  selectedOption: { id: number; title: string } | null;
+  onSelect: (goal: { id: number; title: string } | null) => void;
   placeholder: string;
-  isDropdownOpen: boolean;
-  onToggle: () => void;
 }
 
-function Dropdown({
-  options,
-  selectedOption,
-  onSelect,
-  placeholder,
-  isDropdownOpen,
-  onToggle,
-}: DropdownProps) {
+function Dropdown({ selectedOption, onSelect, placeholder }: DropdownProps) {
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage, isLoading } =
+    useGetGoals();
+  const options =
+    data?.pages.flatMap((page) =>
+      page.data.goals.map(({ id, title }: Pick<Goal, 'id' | 'title'>) => ({
+        id,
+        title,
+      })),
+    ) || [];
+  const { ref, inView } = useInView();
+  const [isOpen, setIsOpen] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const animationRef = useRef<HTMLDivElement>(null);
+
+  const handleToggleDropdown = () => {
+    setIsOpen((prevIsOpen) => !prevIsOpen);
+  };
+
+  useEffect(() => {
+    if (inView && hasNextPage && !isFetchingNextPage) {
+      fetchNextPage();
+    }
+  }, [inView, hasNextPage, isFetchingNextPage, fetchNextPage]);
+
+  useEffect(() => {
+    if (isOpen && hasNextPage && !isFetchingNextPage) {
+      fetchNextPage();
+    }
+  }, [isOpen, hasNextPage, isFetchingNextPage, fetchNextPage]);
```

Comment:
```suggestion
  useEffect(() => {
    if ((inView || isOpen) && hasNextPage && !isFetchingNextPage) {
      fetchNextPage();
    }
  }, [inView, isOpen, hasNextPage, isFetchingNextPage, fetchNextPage]);
```
[Suggestion] 이런식으로 쓸 수 있을까용??